app/utils/generate-pdf.js

```diff
@@ -0,0 +1,56 @@
+async function generateDecisionReport(reportId) {
+  const response = await fetch(`/generate-decision-report/${reportId}`);
+  try {
+    const data = await response.json();
+    if (!response.ok) {
+      throw new Error(
+        `Backend response contained an error (status: ${response.status}): ${JSON.stringify(data)}`
+      );
+    }
+    return data
+  } catch (error) {
+    // Errors returned from services *should* still
+    // be valid JSON(:API), but we could encounter
+    // non-JSON if e.g. a service is down. If so,
+    // throw a nice error that only contains the
+    // response status.
+    if (error instanceof SyntaxError) {
+      throw new Error(
+        `Backend response contained an error (status: ${response.status})`
+      );
+    } else {
+      throw error;
+    }
+  }
+}
```

Comment:
@ValenberghsSven I added these util functions before you added the `DecisionReportGeneration` service. I will start using the service in this PR as well, but I'm wondering if we shouldn't use the error handling here in the service? Right now the toaster (like the component code it got extracted from) only shows a toast with a generic error if something goes wrong, showing the error from the service might make debugging actual issues easier?

Reply:
Made the service to avoid duplication, any error handling is welcome

Reply:
Added here https://github.com/kanselarij-vlaanderen/frontend-kaleidos/pull/1869/commits/9809bae3f2652520d1d1f07f7ba27973f39ee6ad